kernel/src/table_properties.rs

```diff
@@ -0,0 +1,373 @@
+
+    /// How long the history for a Delta table is kept.
+    ///
+    /// Each time a checkpoint is written, Delta Lake automatically cleans up log entries older
+    /// than the retention interval. If you set this property to a large enough value, many log
+    /// entries are retained. This should not impact performance as operations against the log are
+    /// constant time. Operations on history are parallel but will become more expensive as the log
+    /// size increases.
+    #[serde(rename = "delta.logRetentionDuration")]
+    #[serde(deserialize_with = "deserialize_interval")]
+    pub log_retention_duration: Option<Duration>,
+
+    /// TODO docs
+    #[serde(rename = "delta.enableExpiredLogCleanup")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub enable_expired_log_cleanup: Option<bool>,
+
+    // /// The minimum required protocol reader version for a reader that allows to read from this Delta table.
+    // #[serde(rename = "delta.minReaderVersion")]
+    // pub min_reader_version: Option<u8>,
+
+    // /// The minimum required protocol writer version for a writer that allows to write to this Delta table.
+    // #[serde(rename = "delta.minWriterVersion")]
+    // pub min_writer_version: Option<u8>,
+    //
+    /// true for Delta to generate a random prefix for a file path instead of partition information.
+    ///
+    /// For example, this may improve Amazon S3 performance when Delta Lake needs to send very high
+    /// volumes of Amazon S3 calls to better partition across S3 servers.
+    #[serde(rename = "delta.randomizeFilePrefixes")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub randomize_file_prefixes: Option<bool>,
+
+    /// When delta.randomizeFilePrefixes is set to true, the number of characters that Delta
+    /// generates for random prefixes.
+    #[serde(rename = "delta.randomPrefixLength")]
+    #[serde(deserialize_with = "deserialize_pos_int")]
+    pub random_prefix_length: Option<u64>,
+
+    /// The shortest duration within which new snapshots will retain transaction identifiers (for
+    /// example, SetTransactions). When a new snapshot sees a transaction identifier older than or
+    /// equal to the duration specified by this property, the snapshot considers it expired and
+    /// ignores it. The SetTransaction identifier is used when making the writes idempotent.
+    #[serde(rename = "delta.setTransactionRetentionDuration")]
+    #[serde(deserialize_with = "deserialize_interval")]
+    pub set_transaction_retention_duration: Option<Duration>,
+
+    /// The target file size in bytes or higher units for file tuning. For example, 104857600
+    /// (bytes) or 100mb.
+    ///
+    /// TODO: kernel doesn't govern file writes? should we pass this through just with a note that
+    /// we don't pay attention to it? Scenario: engine calls
+    /// snapshot.table_properties().target_file_size in order to know how big _it_ should write
+    /// parquet files.
```

Comment:
That reasoning seems correct? Engine is welcome to check that property and do what it wants, but kernel's own default engine parquet writer should ideally honor that property, if arrow-rust makes that reasonably easy to do so.

Reply:
cool i'll delete this thanks!